File: ted-ext/ted-scheduler/src/test/java/ted/scheduler/TestConfig.java
package ted.scheduler;

import com.zaxxer.hikari.HikariDataSource;
import ted.driver.Ted.TedDbType;

import javax.sql.DataSource;

/**
 * @author Augustus
 *         created on 2016.09.22
 */
class TestConfig {
	static final boolean INT_TESTS_ENABLED = true;
	static final String SYSTEM_ID = "ted.test";
	static final TedDbType testDbType = TedDbType.POSTGRES; // which one we are testing

	private static DataSource dataSource = null;

	static DataSource getDataSource() {
		synchronized (TestConfig.class) {
			if (dataSource == null)
				dataSource = dataSource();
		}
		return dataSource;
	}

	private static DataSource dataSource() {
		HikariDataSource dataSource = new HikariDataSource();
		dataSource.setDriverClassName("org.postgresql.Driver");
		dataSource.setJdbcUrl("jdbc:postgresql://localhost:5433/ted");
		dataSource.setUsername("ted");
		dataSource.setPassword("ted");
		return dataSource;
	}

}

File: ted-ext/ted-scheduler/src/test/java/ted/scheduler/TestConfig.kt
package ted.scheduler

import com.zaxxer.hikari.HikariDataSource
import ted.driver.Ted.TedDbType

import javax.sql.DataSource

/**
 * @author Augustus
 * created on 2016.09.22
 */
internal object TestConfig {
    val INT_TESTS_ENABLED = true
    val SYSTEM_ID = "ted.test"
    val testDbType = TedDbType.POSTGRES // which one we are testing

    private var dataSource: DataSource? = null

    fun getDataSource(): DataSource {
        synchronized(TestConfig::class.java) {
            if (dataSource == null)
                dataSource = dataSource()
        }
        return dataSource!!
    }

    private fun dataSource(): DataSource {
        val dataSource = HikariDataSource()
        dataSource.driverClassName = "org.postgresql.Driver"
        dataSource.jdbcUrl = "jdbc:postgresql://localhost:5433/ted"
        dataSource.username = "ted"
        dataSource.password = "ted"
        return dataSource
    }

}
